tedana/model/fit.py

```diff
@@ -8,77 +8,71 @@
 import pandas as pd
 from scipy import stats
 
-from tedana import (combine, io, utils)
+from tedana import io, utils
 
 LGR = logging.getLogger(__name__)
 
 F_MAX = 500
 Z_MAX = 8
 
 
-def fitmodels_direct(catd, mmix, mask, t2s, t2s_full, tes, combmode, ref_img,
-                     reindex=False, mmixN=None, full_sel=True, label=None,
-                     out_dir='.', verbose=False):
+def dependence_metrics(catd, tsoc, mmix, mask, t2s, tes, ref_img,
+                       reindex=False, mmixN=None, method=None, label=None,
+                       out_dir='.', verbose=False):
     """
     Fit TE-dependence and -independence models to components.
 
     Parameters
     ----------
     catd : (S x E x T) array_like
         Input data, where `S` is samples, `E` is echos, and `T` is time
+    tsoc : (S x T) array_like
+        Optimally combined data
     mmix : (T x C) array_like
         Mixing matrix for converting input data to component space, where `C`
         is components and `T` is the same as in `catd`
     mask : (S [x E]) array_like
         Boolean mask array
     t2s : (S [x T]) array_like
         Limited T2* map or timeseries.
-    t2s_full : (S [x T]) array_like
-        Full T2* map or timeseries. For voxels with good signal in only one
-        echo, which are zeros in the limited T2* map, this map uses the T2*
-        estimate using the first two echoes.
     tes : list
         List of echo times associated with `catd`, in milliseconds
-    combmode : {'t2s', 'ste'} str
-        How optimal combination of echos should be made, where 't2s' indicates
-        using the method of Posse 1999 and 'ste' indicates using the method of
-        Poser 2006
     ref_img : str or img_like
         Reference image to dictate how outputs are saved to disk
     reindex : bool, optional
-        Default: False
-    mmixN : array_like, optional
-        Default: None
-    full_sel : bool, optional
-        Whether to perform selection of components based on Rho/Kappa scores.
-        Default: True
+        Whether to sort components in descending order by Kappa. Default: False
+    mmixN : (T x C) array_like, optional
+        Z-scored mixing matrix. Default: None
+    method : {'kundu_v2', 'kundu_v3', None}, optional
```

Comment:
We calculate some metric maps that are only used in v3.2, like `WTS`, `tsoc_B`, `PSC`, and `F_S0_maps`. Those maps are currently sort of vestigial, but I didn't want to remove them completely in case we do add v3 back in.